### PR TITLE
Rename Pool to IPPool

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

**Comment by robbrockbank** [comment]
@gunjan5 Can you review this but please don't merge just yet as I'd like to get the calico-containers updated as well, but the PR is backed up against another.

File: lib/backend/model/ippool.go [hunk 1]
```diff
@@ -27,15 +27,15 @@ import (
 )
 
 var (
-	matchPool = regexp.MustCompile("^/?calico/v1/ipam/v./pool/([^/]+)$")
-	typePool  = reflect.TypeOf(Pool{})
+	matchIPPool = regexp.MustCompile("^/?calico/v1/ipam/v./pool/([^/]+)$")
```
Comment (gunjan5) [inline lib/backend/model/ippool.go]: this shouldn't matter but, are we not changing the etcd key path from pool to ippool?

File: lib/backend/model/ippool.go [hunk 2]
```diff
@@ -44,27 +44,27 @@ func (key PoolKey) defaultPath() (string, error) {
...
 
-func (key PoolKey) valueType() reflect.Type {
-	return typePool
+func (key IPPoolKey) valueType() reflect.Type {
+	return typeIPPool
 }
 
-func (key PoolKey) String() string {
+func (key IPPoolKey) String() string {
 	return fmt.Sprintf("Pool(cidr=%s)", key.CIDR)
```
Comment (gunjan5) [inline lib/backend/model/ippool.go]: shouldn't this be IPPool too?

File: lib/client/pool.go [hunk 3]
```diff
@@ -21,87 +21,87 @@ import (
 )
 
 // PoolInterface has methods to work with Pool resources.
```
Comment (gunjan5) [inline lib/client/pool.go]: PoolInterface -> IPPoolInterface, Pool -> IPPool

File: lib/testutils/client.go [hunk 4]
```diff
@@ -47,23 +47,23 @@ func NewClient(cf string) (*client.Client, error) {
 
 // CreateNewPool takes a client.Client with a poolSubnet CIDR (in "192.168.1.0/24" format) with
```
Comment (gunjan5) [inline lib/testutils/client.go]: CreateNewPool -> CreateNewIPPool

File: lib/backend/model/ippool.go (hunk 1 above)
Comment (caseydavenport) [inline lib/backend/model/ippool.go]: nope, the etcdv2 key is a "stable" api itself, and has to be compatible with the python version (which uses `pool`)

File: lib/client/pool_e2e_test.go [hunk 5]
```diff
@@ -60,27 +60,27 @@ var _ = Describe("Pool tests", func() {
 				log.Println("Error creating client:", err)
 			}
 			By("Updating the pool before it is created")
-			_, outError := c.Pools().Update(&api.Pool{Metadata: meta1, Spec: spec1})
+			_, outError := c.IPPools().Update(&api.IPPool{Metadata: meta1, Spec: spec1})
 
 			// Should return an error.
 			Expect(outError.Error()).To(Equal(errors.New("resource does not exist: Pool(cidr=10.0.0.0/24)").Error()))
```
Comment (gunjan5) [inline lib/client/pool_e2e_test.go]: one last thing, if you end up changing https://github.com/projectcalico/libcalico-go/pull/240#discussion-diff-86075055R60, this error assertion would also change along with Line 147 in this file

**Comment by robbrockbank** [comment]
Changes applied - and a few more other comments fix ups that I noticed.

**Comment by robbrockbank** [comment]
Changes applied - merging

**Pull request merged**